string book_name
string bin_name
float64 x
float64 y
string orientation
---
float64 result
